src/containers/edit-profile/professional-info-tab/ProfessionalInfoTab.tsx

```diff
@@ -31,110 +29,36 @@ import AppButton from '~/components/app-button/AppButton'
-    isActivationBlocked: true,
-    subjects: [
-      {
-        _id: crypto.randomUUID(),
-        name: 'English',
-        proficiencyLevels: [
-          ProficiencyLevelEnum.Beginner,
-          ProficiencyLevelEnum.Intermediate,
-          ProficiencyLevelEnum.Advanced,
-          ProficiencyLevelEnum.Professional,
-          ProficiencyLevelEnum.TestPreparation
-        ]
-      },
-      {
-        _id: crypto.randomUUID(),
-        name: 'English',
-        proficiencyLevels: [
-          ProficiencyLevelEnum.Beginner,
-          ProficiencyLevelEnum.Intermediate,
-          ProficiencyLevelEnum.Advanced
-        ]
-      }
-    ]
-  },
-  {
-    _id: crypto.randomUUID(),
-    name: 'Languages',
-    isActivated: true,
-    isActivationBlocked: false,
-    subjects: [
-      {
-        _id: crypto.randomUUID(),
-        name: 'English',
-        proficiencyLevels: [
-          ProficiencyLevelEnum.Beginner,
-          ProficiencyLevelEnum.Intermediate,
-          ProficiencyLevelEnum.Advanced,
-          ProficiencyLevelEnum.Professional,
-          ProficiencyLevelEnum.TestPreparation
-        ]
-      },
-      {
-        _id: crypto.randomUUID(),
-        name: 'English',
-        proficiencyLevels: [
-          ProficiencyLevelEnum.Beginner,
-          ProficiencyLevelEnum.Intermediate,
-          ProficiencyLevelEnum.Advanced
-        ]
-      }
-    ]
-  },
-  {
-    _id: crypto.randomUUID(),
-    name: 'Computer science',
-    isActivated: false,
-    isActivationBlocked: false,
-    subjects: [
-      {
-        _id: crypto.randomUUID(),
-        name: 'PHP',
-        proficiencyLevels: [
-          ProficiencyLevelEnum.Beginner,
-          ProficiencyLevelEnum.Intermediate,
-          ProficiencyLevelEnum.Advanced
-        ]
-      },
-      {
-        _id: crypto.randomUUID(),
-        name: 'Java',
-        proficiencyLevels: [ProficiencyLevelEnum.Beginner]
-      }
-    ]
-  }
-]
-
 interface ProfessionalInfoTabProps {
   professionalBlock?: ProfessionalBlock
-  categories: SubjectInterface[]
-  userId: string
+  categories: UserMainSubject[]
 }
 
 const ProfessionalInfoTab: FC<ProfessionalInfoTabProps> = ({
   professionalBlock,
-  userId
+  categories = []
 }) => {
   const { t } = useTranslation()
 
-  const { openModal, closeModal } = useModalContext()
+  const { userId, userRole } = useAppSelector((state) => state.appMain)
 
-  const { userRole } = useAppSelector((state) => state.appMain)
+  const { openModal, closeModal } = useModalContext()
 
-  const { handleSubmit, loading } = useUpdateUser(userId)
+  const { handleSubmit, loading } = useUpdateUser(userId, true)
 
   const { data, handleInputChange } = useForm<ProfessionalBlock>({
     initialValues: professionalBlock || initialFormValues
   })
 
+  const handleDeleteCategory = (mainSubjectId: string, categoryId: string) => {
+    handleSubmit({
+      mainSubjects: {
+        _id: mainSubjectId,
+        category: { _id: categoryId, name: '' }
```

Comment:
same here. Think about move the logic to a separate variable

Reply:
Thank you fixed!